calculator.go

```diff
@@ -1,13 +1,96 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
+import (
+	"go/token"
+	"go/types"
+	"math"
+	"strconv"
+)
+
+const divideByZeroMessage = "Cannot divide by zero"
+const negativeNumber = "No negative number allowed"
+
+// ErrorString is a trivial implementation of error.
+type ErrorString struct {
```

Comment:
And we don't have to implement this ourselves, because the standard library can help here: `errors.New` takes a string and constructs an `error` value which returns that string.